src/Microsoft.OData.Client/Materialization/FeedAndEntryMaterializerAdapter.cs

```diff
@@ -420,6 +547,49 @@ private ODataNestedResourceInfo ReadNestedResourceInfo()
             return link;
         }
 
+        /// <summary>
+        /// Reads an <see cref="ODataDeletedResource"/>
+        /// </summary>
+        /// <returns>An <see cref="MaterializerDeletedEntry"/> of the read <see cref="ODataDeletedResource"/>.</returns>
+        private MaterializerDeletedEntry ReadDeletedResource()
+        {
+            this.ExpectState(ODataReaderState.DeletedResourceStart);
+
+            ODataDeletedResource result = (ODataDeletedResource)this.reader.Item;
+            MaterializerDeletedEntry entry;
+
+            Debug.Assert(this.items != null, "this.items should not be null");
+
+            entry = MaterializerDeletedEntry.CreateDeletedEntry(
+                    result,
+                    this.materializerContext);
+
+            if (this.items.Count > 0)

```

Comment:
Do we do this check because a deleted resource cannot be at the top-level?
If yes, shouldn't we have an `else` block where we throw an exception if we find ourselves with a deleted resource in our hands and there's no top-level item in the stack?

Reply:
Yes. A deleted resource in a bulk update can be top-level but it should not be the first item in the stack. The first item has to be a `MaterializerDeltaFeed` set to tell us that we are doing a bulk update...So if we find ourselves adding a deleted resource to an empty stack that is a problem. I could add the if--else but I don't think we'll ever get to the else part in a bulk operation or I could remove the check all together.

Reply:
@ElizabethOkerio Exactly. Remove the check altogether. The logic as it is suggests that such a scenario is possible and that we'd just ignore it and move on... You could add a `Debug.Assert(this.items.Count > 0, "yada yada");`

Reply:
I've removed the check also there is already a Debug.Assert